addons/microsoft_outlook/views/ir_mail_server_views.xml

```diff
@@ -8,6 +8,19 @@
             <field name="smtp_host" position="before">
                 <field name="use_microsoft_outlook_service" string="Outlook"/>
             </field>
+            <xpath expr="//field[@name='smtp_user']" position="attributes">
+                <attribute name="attrs">
+                    {'required' : [('use_microsoft_outlook_service','=', True)] }
+                </attribute>
+                <attribute name="placeholder">
+                    Outlook/Office365 email address
+                </attribute>
```

Comment:
Other than a generic placeholder, we could have a different field and hide them conditionally to display different placeholders...

But I think just a more generic placeholder is enough for this ^^

Reply:
Tried to use t-if etc. but didn't work as expected. Will try some things and worst we just keep it generic. That being said, relating to the fact that the gmail module might have the same issue, I don't know if the xpath approach is recommend.

When both modules are installed (outlook and gmail), it seems that the changes get overwritten during view inheritance. Still need to understand what's going on.

Reply:
Relating to case where a user install both "google_gmail" and "microsoft_outlook":

- In google_gmail, we find in  `ir_mail_server_views.xml` 
```xml
           <field name="smtp_pass" position="attributes">
                <attribute name="attrs">{'invisible' : [('use_google_gmail_service', '=', True)]}</attribute>
            </field>
``` 
which defacto replaces the attributes with the domain in question

-   But if inmy modified `ir_mail_server_views.xml` for microsoft_outlook, I use 
```xml
            <xpath expr="//field[@name='smtp_pass']" position="attributes">
                <attribute name="attrs">{'invisible' : [('use_microsoft_outlook_service','=', True)] }
                </attribute>
            </xpath>
```
Which depending on how extended views get loaded, leads us to have `attrs="{'invisible' : [('use_microsoft_outlook_service','=', True)] }"` or `attrs="{'invisible' : [('use_google_gmail_service', '=', True)]}"`, but it seems that it can't append to the domain, i.e. ideally `attrs="{'invisible' : ['|', ('use_google_gmail_service', '=', True), ('use_microsoft_outlook_service','=', True)]}"`

Perhaps this is not possible with the current framework and ROI in finding a workaround potentially not worth it. But if you have any ideas how to correctly hide the password field when either use_outlook or use_gmail boxes are ticked in the form view, I would love to know a solution ;)